store/v2/snapshots/store.go

```diff
@@ -172,14 +172,15 @@ func (s *Store) Load(height uint64, format uint32) (*types.Snapshot, <-chan io.R
 				_ = pw.CloseWithError(err)
 				return
 			}
-			defer chunk.Close()
-			_, err = io.Copy(pw, chunk)
-			if err != nil {
-				_ = pw.CloseWithError(err)
-				return
-			}
-			chunk.Close()
-			pw.Close()
+			func() {
+				defer chunk.Close()
+				_, err = io.Copy(pw, chunk)
+				if err != nil {
+					_ = pw.CloseWithError(err)
+					return
+				}
+				pw.Close()
+			}()
```

Comment:
```suggestion
			{
				defer chunk.Close()
				_, err = io.Copy(pw, chunk)
				if err != nil {
					_ = pw.CloseWithError(err)
					return
				}
				pw.Close()
			}
```

This illustrates what you are trying to do a bit more concisely - we just need this code in its own scope

Reply:
This code snippet was wrapped in a `for range`.

More details info: https://go.dev/play/p/AKEDPEnWaAa

![图片](https://github.com/user-attachments/assets/f93f0b99-f552-42f1-a77d-bb6176ed6186)

Reply:
ah I see - thanks for pointing this out !